app/controllers/nodes_controller.rb

```diff
@@ -22,14 +22,16 @@ def show
   # assigned role, assign a random role to it, and then call the salt
   # orchestration.
   def bootstrap
-    available_roles = [:master]
+    @available_roles = [:master]
```

Comment:
The use of an instance variable here is a little cryptic (this is only needed for the tests). I think it would be better if we changed the signature of this method to:

```
def bootstrap(roles=[:master])
  ...
  return assigned_roles
end
```
This way you don't need to change the method calls and you can check the return value for assigned roles.